tmail-backend/saas/saas-api/src/main/java/com/linagora/tmail/saas/api/SaaSUserRepository.java

```diff
@@ -0,0 +1,30 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ *******************************************************************/
+
+package com.linagora.tmail.saas.api;
+
+import org.apache.james.core.Username;
+import org.reactivestreams.Publisher;
+
+import com.linagora.tmail.saas.model.SaaSPlan;
+
+public interface SaaSUserRepository {
+    Publisher<SaaSPlan> getPlan(Username username);
+
+    Publisher<Void> setPlan(Username username, SaaSPlan saaSPlan);
```

Comment:
IMO I'd rather Create/update then read a POJO called `SaaSAccount` with (for now?) a single property being `SaaSPlan`